presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/cli/EnvironmentUp.java

```diff
@@ -139,18 +147,14 @@ private void killContainersReaperContainer()
             }
         }
 
-        private void sleepUntilInterrupted()
+        private void waitUntil(CheckedSupplier<Boolean> completed)
```

Comment:
This method could accept `Predicate`. Then it will be clear that this method will wait until given predicate. It is not that clear with `CheckedSupplier<Boolean>`. Also using `Predicate` hides implementation detail of `failsafe` usage. Wrapping `Predicate` to `CheckedSupplier<Boolean>` should be trivial.

Reply:
https://github.com/prestosql/presto/pull/3897#discussion_r433657772